src/Common/Commands.Common/PowerBICmdlet.cs

```diff
@@ -179,13 +180,71 @@ protected bool InteractiveConsole
 
         protected virtual bool IsTerminatingError(Exception ex) => ex is PipelineStoppedException pipelineStoppedEx && pipelineStoppedEx.InnerException == null || ex is NotImplementedException;
 
-        protected virtual string CurrentPath => this.SessionState != null ? SessionState.Path.CurrentLocation.Path : Directory.GetCurrentDirectory();
+        protected virtual string CurrentPath => this.SessionState != null ? this.SessionState.Path.CurrentLocation.Path : Directory.GetCurrentDirectory();
+
+        protected virtual string ResolveFilePath(string path, bool isLiteralPath)
+        {
+            ProviderInfo provider = null;
+            PSDriveInfo drive = null;
+            var filePaths = new List<string>();
+
+            if (this.SessionState != null)
+            {
+                try
+                {
+                    if (isLiteralPath)
+                    {
+                        filePaths.Add(this.SessionState.Path.GetUnresolvedProviderPathFromPSPath(path, out provider, out drive));
+                    }
+                    else
+                    {
+                        filePaths.AddRange(this.SessionState.Path.GetResolvedProviderPathFromPSPath(path, out provider));
+                    }
+
+                    if (provider != null && !provider.Name.Contains("FileSystem"))
```

Comment:
Does this need to be case-insensitive?

Reply:
Probably doesn't hurt if it was case-insensitive